src/main/java/org/opensearch/security/dlic/rest/api/ApiTokenApiAction.java

```diff
@@ -0,0 +1,114 @@
+import static org.opensearch.security.dlic.rest.api.Responses.*;
+import static org.opensearch.security.dlic.rest.support.Utils.addRoutesPrefix;
+import static org.opensearch.security.securityconf.impl.v7.ConfigV7.*;
+
+public class ApiTokenApiAction extends AbstractApiAction {
+
+    public static final String NAME_JSON_PROPERTY = "ip";
+
+
+    private static final List<Route> ROUTES = addRoutesPrefix(
+        ImmutableList.of(
+            new Route(GET, "/apitokens"),
+            new Route(PUT, "/apitokens/{name}")
+//            new Route(DELETE, "/apitokens/{name}"),
+        )
+    );
+
+    protected ApiTokenApiAction(ClusterService clusterService, ThreadPool threadPool, SecurityApiDependencies securityApiDependencies) {
+        super(Endpoint.APITOKENS, clusterService, threadPool, securityApiDependencies);
+        this.requestHandlersBuilder.configureRequestHandlers(this::authFailureConfigApiRequestHandlers);
+    }
+
+    @Override
+    public String getName() {
+        return "API Token actions to retrieve / update configs.";
+    }
+
+    @Override
+    public List<Route> routes() {
+        return ROUTES;
+    }
+
+    @Override
+    protected CType<ConfigV7> getConfigType() {
+        return CType.CONFIG;
+    }
+
+    private void authFailureConfigApiRequestHandlers(RequestHandler.RequestHandlersBuilder requestHandlersBuilder) {
+
+        requestHandlersBuilder.override(
+            GET,
+            (channel, request, client) -> loadConfiguration(getConfigType(), false, false).valid(configuration -> {
+                if (!apiTokenIndexExists()) {
+                    ok(channel, "empty list");
+                } else {
+                    ok(channel, "non-empty list");
+                }
+            }).error((status, toXContent) -> response(channel, status, toXContent)))
+        .override(PUT, (channel, request, client) -> loadConfiguration(getConfigType(), false, false).valid(configuration -> {
+                String token = createApiToken(request.param(NAME_JSON_PROPERTY), client);
+                ok(channel, token + " created successfully");
+            }).error((status, toXContent) -> response(channel, status, toXContent)));
+
+    }
+
+    public String createApiToken(String name, Client client) {
+        createApiTokenIndexIfAbsent(client);
+
+        return "test-token";
+    }
+
+    public Boolean apiTokenIndexExists() {
+        return clusterService.state().metadata().hasConcreteIndex(ConfigConstants.OPENSEARCH_API_TOKENS_INDEX);
+    }
+
+    public void createApiTokenIndexIfAbsent(Client client) {
+        if (!apiTokenIndexExists()) {
+            final Map<String, Object> indexSettings = ImmutableMap.of("index.number_of_shards", 1, "index.auto_expand_replicas", "0-all");
+            final CreateIndexRequest createIndexRequest = new CreateIndexRequest(ConfigConstants.OPENSEARCH_API_TOKENS_INDEX).settings(indexSettings);
```

Comment:
FYI By subclassing AbstractApiAction this would be authorized like other security APIs. Like other API handlers, I think we will want to go to the transport layer here and execute a transport action and create the index from within there. 

Any calls to a system index should be wrapped into threadContext.stashContext to assure that the plugin can perform the action regardless of the authenticated user's permissions.

Reply:
Good call out. Stashed the context in: [064ffa4](https://github.com/opensearch-project/security/pull/4912/commits/064ffa4a45fdfc13ae7a3515fa941567f6a88a9d)